lib/src/components/form/number_input.dart

```diff
@@ -154,16 +154,17 @@ class _NumberInputState extends State<NumberInput> {
                         },
                       ),
                       enabled: widget.enabled ??
-                          (widget.min == null || _lastValidValue > widget.min!),
+                          (widget.max == null || _lastValidValue < widget.max!),
                       onPressed: () {
-                        if (widget.min == null ||
-                            _lastValidValue > widget.min!) {
+                        if (widget.max == null ||
+                            _lastValidValue < widget.max!) {
                           double oldValue = _value.toDouble();
                           _lastValidValue = oldValue + widget.step;
                           _controller.text = widget.allowDecimals
                               ? _lastValidValue.toString()
                               : _lastValidValue.toInt().toString();
                           widget.onChanged?.call(_lastValidValue);
+                          setState(() {});
```

Comment:
Is rebuilding the widget necessary here?

Reply:
disabled button not enabling and enabled button disabling without rebuilding